ADVANCED.md

```diff
@@ -501,23 +501,6 @@ You may also want to use Discriminated Unions, please check out [Expressive Reac
 
 Here is a brief intuition for **Discriminated Union Types**:
 
-```ts
-type UserTextEvent = { value: string; target: HTMLInputElement };
-type UserMouseEvent = { value: [number, number]; target: HTMLElement };
-type UserEvent = UserTextEvent | UserMouseEvent;
-function handle(event: UserEvent) {
-  if (typeof event.value === "string") {
-    event.value; // string
-    event.target; // HTMLInputElement | HTMLElement (!!!!)
-    return;
-  }
-  event.value; // [number, number]
-  event.target; // HTMLInputElement | HTMLElement (!!!!)
-}
-```
-
-Even though we have narrowed based on `event.value`, the logic doesn't filter up and sideways to `event.target`. This is because a union type `UserTextEvent | UserMouseEvent` could be BOTH at once. So TypeScript needs a better hint. The solution is to use a literal type to tag each case of your union type:
```

Comment:
I think the example should not be deleted. It's a good illustration. But you are correct, the language here does sound confusing.

You can move it below the actual example and collapse it.
You can change the explanation to "
Take care: TypeScript does not narrow the type of a Discriminated Union on the basis of `typeof` checks. The type guard has to be on the value of a key and not it's type.
https://github.com/microsoft/TypeScript/issues/30506#issuecomment-474858198
"